### PR TITLE
Fix setup crash on newer device interface type codes

### PR DESCRIPTION
**Pull request by giuseppe99barchetta**
## Summary

Fixes a setup crash when GL-iNet routers, such as the BE9300, report newer client interface type codes.

The previous implementation used `list(DeviceInterfaceType)[type]` to map router API type indexes to enum values. Because `UNKNOWN2` has the same enum value as `UNKNOWN`, Python treats it as an alias and omits it from `list(DeviceInterfaceType)`. This shifts/collapses the index mapping and can raise `IndexError` when newer routers report higher type codes.

This change adds an explicit interface type mapping tuple that preserves the router API index positions, plus a safe helper that falls back to `UNKNOWN` for invalid or out-of-range values.

## Testing

Tested manually on Home Assistant with a GL-iNet BE9300. The integration setup now completes successfully and the integration works again with this fix.

### CODE REVIEW

**Comment by HarvsG** [comment]
Thanks for the PR, will review in a couple of weeks when life is less busy. Feel free to nudge me if I forget

File: custom_components/glinet/router.py [hunk 1]
```diff
@@ -67,6 +67,36 @@ class DeviceInterfaceType(StrEnum):
     WIFI_6_GUEST = "6GHz Guest"
 
 
+DEVICE_INTERFACE_TYPES: tuple[DeviceInterfaceType, ...] = (
```
Comment (HarvsG) [inline custom_components/glinet/router.py]: What is the purpose of this tuple?

Comment (giuseppe99barchetta) [reply custom_components/glinet/router.py]: Hey, thanks for your time! Hope to see a fix soon, as this issue is currently blocking support for newer GL-iNet routers.

To answer your question:

The GL-iNet API returns a numeric `type` value (`0`, `1`, `2`, ...), so we need a structure where index `N` always maps to API code `N`.

Using the enum directly doesn't work because `UNKNOWN` and `UNKNOWN2` both have the value `"Unknown"`. Python treats `UNKNOWN2` as an alias of `UNKNOWN`, so it disappears when iterating over the enum.

That shifts all indices after position `8`, causing some API type codes to resolve incorrectly and raise an `IndexError`.

The tuple simply preserves the original API index mapping, including both positions `5` and `8`, even though they currently map to the same enum value.

Let me know if I can clarify anything else! Thanks again for your work!

Comment (HarvsG) [reply custom_components/glinet/router.py]: Cool, I don't think we need both this and the strEnum? Can you consolidate into one class with a method that safely handles out of range access?

Also it would be great to log a warning or info for out of range indices that includes some device model and some key dev_info with a note to open an issue so I can continue to expand the class with accurate info over time

Comment (giuseppe99barchetta) [reply custom_components/glinet/router.py]: Good point! I consolidated the mapping into `DeviceInterfaceType` by adding `from_index()` class method.

It now uses `__members__`, which preserves aliases and declaration order, unlike enum. Invalid, non-numeric, or out-of-range values safely fall back to `UNKNOWN`.

I also added a warning containing the router model, firmware version, online state, and available `dev_info` fields, with a request to open an issue.

Let me know!

**Comment by shauneccles** [comment]
Thanks for this — having read through it, it's a solid fix. Using `cls.__members__` to preserve the enum index order (so `WIFI_6_GUEST` at index 12 resolves correctly while keeping the duplicate member) is neat, and the log-once `_log_unsupported` with model/firmware context is a nice touch for collecting unknown codes from the field.

One robustness suggestion: the router also returns a self-describing `iface` string per client (`"2.4G"`, `"5G"`, `"6G"`, `"cable"`, …) alongside the integer `type`. Resolving from that string first — and falling back to the integer index, as you already do — makes interface labelling resilient to firmware that renumbers the `type` codes; the integer-index approach silently mislabels if that order ever changes. Removing the duplicate `UNKNOWN2 = "Unknown"` member outright (rather than relying on `__members__` ordering) also closes the `list(enum)` foot-gun that caused the original bug for good.

I prototyped that approach here if any of it is useful to fold in: https://github.com/HarvsG/ha-glinet4-integration/compare/master...shauneccles:ha-glinet4-integration:pr/wifi7-crash — no worries either way, your PR fixes the crash; I'm just flagging the `iface`-string angle as belt-and-braces.

For context, I've opened two follow-ups that build on the crash being fixed: #146 (device-tracker discovery + randomized-MAC handling, #139) and #147 (dynamic WiFi-interface switches, #103). They deliberately leave interface-type resolution to this PR.

**Review by HarvsG — CHANGES_REQUESTED** [review]
I would be grateful for an update considering the above and resolving the merge conflict

**Comment by giuseppe99barchetta** [comment]
I’ve updated the branch to the latest master and resolved the merge conflict.

I also incorporated the suggestion above: interface detection now prefers the self-describing iface value when available, with the numeric type kept as a fallback for compatibility.

I added tests covering the different interface values and fallback cases as well.

It should now be ready for another review. Thanks again!

**Comment by HarvsG** [comment]
Hopefully fixes #143

**Comment by giuseppe99barchetta** [comment]
> Hopefully fixes #143

Yes, that should be covered as well. The explicit numeric mapping already handles `type=12` as `WIFI_6_GUEST`, and with this update the `iface` value is preferred when available.

I also added a test specifically for `6G Guest` to make sure it resolves correctly even if the numeric type value is different.

File: custom_components/glinet/router.py [hunk 2]
```diff
@@ -99,6 +99,45 @@ class DeviceInterfaceType(StrEnum):
...
+            if marker in iface:
+                return guest_type
+    elif "mlo" in iface:
+        return DeviceInterfaceType.MLO
+
+    raw_type = dev_info.get("type", 5)
+    try:
+        type_index = -1 if isinstance(raw_type, bool) else int(raw_type)
+    except (TypeError, ValueError):
+        type_index = -1
```
Comment (HarvsG) [inline custom_components/glinet/router.py:145]: In the case of an un-matched iface and a new type_index I would like to log it before we return - something like
```python
if device.interface_type == DeviceInterfaceType.UNKNOWN and unmapped_type not in (5, 8, None):
    if unmapped_type not in self._warned_interface_types:
        self._warned_interface_types.add(unmapped_type)
        _LOGGER.warning(
            "GL-iNet router %s (model: %s, firmware: %s) reported an unhandled client interface "
            "(type=%s, iface=%s, mac=%s). Please open an issue at "
            "https://github.com/HarvsG/ha-glinet4-integration/issues",
            self._host, self._model, self._sw_v, unmapped_type, dev_info.get("iface"), device_mac
        )
```

File: custom_components/glinet/router.py [hunk 3]
```diff
@@ -99,6 +99,45 @@ class DeviceInterfaceType(StrEnum):
...
+    interface_type = DEVICE_INTERFACE_IFACE_MAP.get(iface)
+    if interface_type is not None:
+        return interface_type
+
+    if "guest" in iface:
+        guest_interfaces = (
+            ("2.4", DeviceInterfaceType.WIFI_24_GUEST),
+            ("6", DeviceInterfaceType.WIFI_6_GUEST),
+            ("mlo", DeviceInterfaceType.MLO_GUEST),
+            ("5", DeviceInterfaceType.WIFI_5_GUEST),
```
Comment (HarvsG) [inline custom_components/glinet/router.py]: `5` and `6` are quite broad markers - I wonder if there are GL-inet devices that could have an eth6 or similar that might be incorrectly matched.
If you're not sure of a better way, debug logging matches would be a minimum so we can more easily pick up false matches in future

File: custom_components/glinet/router.py (hunk 2 above)
Comment (HarvsG) [reply custom_components/glinet/router.py:145]: You would need to define `self._warned_interface_types: set[int | str] = set()`

File: custom_components/glinet/router.py [hunk 4]
```diff
@@ -99,6 +99,45 @@ class DeviceInterfaceType(StrEnum):
...
+
+def device_interface_type_from_client(dev_info: dict) -> DeviceInterfaceType:
+    """Resolve a client interface from iface, falling back to the type index."""
+    iface = str(dev_info.get("iface") or "").strip().lower()
+    interface_type = DEVICE_INTERFACE_IFACE_MAP.get(iface)
+    if interface_type is not None:
+        return interface_type
+
+    if "guest" in iface:
+        guest_interfaces = (
```
Comment (HarvsG) [inline custom_components/glinet/router.py]: My AI sidekick thinks this is inefficient and generates lots of dynamic look ups
Apparently it will be faster if you define it as a module-level constant (e.g. `_GUEST_INTERFACE_MARKERS: tuple[tuple[str, DeviceInterfaceType], ...]`

File: custom_components/glinet/router.py [hunk 5]
```diff
@@ -99,6 +99,45 @@ class DeviceInterfaceType(StrEnum):
...
+            ("mlo", DeviceInterfaceType.MLO_GUEST),
+            ("5", DeviceInterfaceType.WIFI_5_GUEST),
+        )
+        for marker, guest_type in guest_interfaces:
+            if marker in iface:
+                return guest_type
+    elif "mlo" in iface:
+        return DeviceInterfaceType.MLO
+
+    raw_type = dev_info.get("type", 5)
```
Comment (HarvsG) [inline custom_components/glinet/router.py]:
```suggestion
    raw_type = dev_info.get("type")
    if raw_type is None:
        raw_type = 5
```
covers the `{type : None}` case

File: tests/test_interface_types.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
+"""Tests for client interface type resolution."""
```
Comment (HarvsG) [inline tests/test_interface_types.py]: Some feedback - please drop the [test_interface_types.py](https://github.com/HarvsG/ha-glinet4-integration/pull/145/changes#diff-053475f87272a6a47f19aaaff09456b1a694d90d7b82da36d91961d204c2fc99) file and if you don't fancy doing the below, I will follow it up in a seperate PR

- File Fragmentation: tests/test_router.py already contains client device and interface mapping tests (lines 145–171). Creating a separate standalone test file test_interface_types.py for helper logic belonging to router.py fragments the test suite.
- Test Duplication: test_numeric_type_fallback directly duplicates tests already present in tests/test_router.py:test_device_interface_type_mapping for index 11 and 99.
- Missing Edge Case Coverage:
   - Missing "iface" or "type" keys (e.g. {} or {"name": "dev"})
   - iface: None and type: None (TypeError branch is currently unexercised)
   - Whitespace handling (e.g. "  2.4G  ")
   - "2.4G Guest" (tested 5G, 6G, and MLO Guest, but omitted 2.4G Guest)
   - "wired" and "lan" strings
   - Substring MLO branch (elif "mlo" in iface:)
   - Direct unit tests for device_interface_type_from_client without requiring ClientDevInfo instantiation.

**Review by HarvsG — CHANGES_REQUESTED** [review]
